hal/inc/socket_hal_posix.h

```diff
@@ -40,6 +40,8 @@ extern "C" {
  *
  */
 
+/** Compatibility nfds_t */
+typedef unsigned int nfds_t;
```

Comment:
We shouldn't be declaring this here. The compatibility types in this section are only for compatibility with non-POSIX socket hal.

`nfds_t` should be coming from the implementation header `socket_hal_posix_impl.h` and in turn from `lwip/sockets.h`.

Reply:
OK, so should I include that header instead?

Reply:
Done

Reply:
It is already included in `socket_hal_posix_impl.h`, so in theory you shouldn't need to add anything at all unless for some reason `POLLIN` or `POLLOUT` are defined by newlib (I don't think they are): https://github.com/particle-iot/lwip/blob/particle/src/include/lwip/sockets.h#L506